components/console/commands.c

```diff
@@ -231,7 +231,12 @@ static const cmd_item_t *find_command_by_name(const char *name)
     return cmd;
 }
 
-esp_err_t esp_console_run(const char *cmdline, int *cmd_ret)
+esp_err_t esp_console_run(const char* cmdline, int* cmd_ret) 
+{
+    return esp_console_run_with_ctx(NULL, cmdline, cmd_ret);
+}
+
+esp_err_t esp_console_run_with_ctx(void* invocation_ctx, const char *cmdline, int *cmd_ret)
```

Comment:
```suggestion
esp_err_t esp_console_run_with_context(void* invocation_ctx, const char *cmdline, int *cmd_ret)
```
Since the type is named `esp_console_cmd_func_with_context_t`, I would suggest to keep `context` instead of `ctx`, even